File: Lab3/src/KI305_Leshkovych_Lab3/VideoRecorder.java
package KI305_Leshkovych_Lab3;

import KI305_Leshkovych_Lab3.VideoPlayer;

/**
 * Клас VideoRecorder реалiзує функцiональнiсть вiдеозаписувача.
 */
public class VideoRecorder extends VideoPlayer implements Recordable {

    /**
     * Записує вiдео з вказаним iм'ям.
     *
     * @param videoName Назва вiдео, яке потрiбно записати.
     */
    @Override
    public void recordVideo(String videoName) {
        log("Запис вiдео: " + videoName);
        System.out.println("Запис вiдео: " + videoName);
    }

    /**
     * Зупиняє запис вiдео.
     */
    @Override
    public void stopRecording() {
        log("Запис зупинено.");
        System.out.println("Запис зупинено.");
    }
}

File: Lab9 — копия/video/video_recorder.py
# video/video_recorder.py

from .video_player import VideoPlayer

class VideoRecorder(VideoPlayer):
    """Похідний клас для запису відео, успадковується від VideoPlayer."""
    
    def __init__(self, video_name):
        """
        Ініціалізація відеомагнітофона з назвою відео.
        :param video_name: Назва відеофайлу
        """
        super().__init__(video_name)
        self.is_recording = False

    def record(self):
        """Метод для початку запису відео."""
        if not self.is_playing:
            print(f"Початок запису відео: {self.video_name}")
            self.is_recording = True
        else:
            print("Неможливо почати запис під час відтворення.")

    def stop_recording(self):
        """Метод для зупинки запису відео."""
        if self.is_recording:
            print(f"Запис зупинено: {self.video_name}")
            self.is_recording = False
        else:
            print("Запис не активний.")